src/ff/field.rs

```diff
@@ -102,14 +92,4 @@ pub trait Field:
     }
 }
 
-pub trait BinaryField:
-    Field
-    + BitAnd<Output = Self>
-    + BitAndAssign
-    + BitOr<Output = Self>
-    + BitOrAssign
-    + BitXor<Output = Self>
-    + BitXorAssign
-    + Not<Output = Self>
-{
-}
+pub trait BinaryField: Field + BooleanOps {}
```

Comment:
I just searched and we don't use `BinaryField` or `Fp2` any more.  Can these be removed?

I started by trying to think why we would want something like this and came up short.  The `BitArray`-based version works better for every case I could think of.

Reply:
Removed Fp2.